### PR TITLE
[ALLUXIO-2193] Move default value to propertyKey from properties file

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-2193

### CODE REVIEW

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 1]
```diff
@@ -652,13 +678,27 @@ public static PropertyKey fromString(String keyStr) {
...
    *
    * @param property String of this property
+   * @param defaultValue Default value of this property in compile time if not null
    */
-  PropertyKey(String property) {
-    mString = property;
+  PropertyKey(String property, Object defaultValue) {
+    mName = property;
+    if (defaultValue == null) {
+      mDefaultValue = null;
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/PropertyKey.java]: Is it expected for property key calls to handle the `null` case?

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 2]
```diff
@@ -606,13 +627,18 @@
         "alluxio.security.group.mapping.class";
     public static final String SECURITY_LOGIN_USERNAME = "alluxio.security.login.username";
 
-    private Name() {} // prevent instantiation
+    private Name() {
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/PropertyKey.java]: undo formatting change?

File: minicluster/src/main/java/alluxio/master/LocalAlluxioMaster.java [hunk 3]
```diff
@@ -66,6 +70,8 @@ public void run() {
         try {
           mAlluxioMaster.start();
         } catch (Exception e) {
+          // LOG the exception as the RuntimeException will be handled silently by JUNIT
```
Comment (jsimsa) [inline minicluster/src/main/java/alluxio/master/LocalAlluxioMaster.java]: JUnit

Comment (jsimsa) [inline minicluster/src/main/java/alluxio/master/LocalAlluxioMaster.java]: should worker code have similar logic?

**Comment by jsimsa** [comment]
@apc999 thanks for the PR ... left a couple of minor comments

File: core/common/src/main/java/alluxio/PropertyKey.java (hunk 1 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: default value of `null` is expected cases here.
some properties are not determined in compile time (e.g., master address), or platform dependent

File: minicluster/src/main/java/alluxio/master/LocalAlluxioMaster.java (hunk 3 above)
Comment (apc999) [inline minicluster/src/main/java/alluxio/master/LocalAlluxioMaster.java]: Fixed

File: core/common/src/main/java/alluxio/PropertyKey.java (hunk 2 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: done

File: core/common/src/main/java/alluxio/PropertyKey.java (hunk 1 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: I see your question. Still in `Configuration.get` if the returned the `null` value, `RuntimeException` is thrown. We didn't change that semantics

**Comment by apc999** [comment]
@jsimsa @calvinjia  PTAL. thanks

**Comment by jsimsa** [comment]
LGTM

File: core/common/src/main/java-templates/alluxio/ProjectConstants.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+package alluxio;
+
+/**
+ * Project constants from compilation time by maven.
+ */
+public final class ProjectConstants {
+  /* Project version, specified in maven property. **/
+  public static final String VERSION = "${project.version}";
```
Comment (aaudiber) [inline core/common/src/main/java-templates/alluxio/ProjectConstants.java]: This is cool

**Comment by aaudiber** [comment]
LGTM

File: core/common/src/main/java-templates/alluxio/ProjectConstants.java (hunk 4 above)
Comment (apc999) [inline core/common/src/main/java-templates/alluxio/ProjectConstants.java]: learnt from @jsimsa and @ifcharming

**Pull request merged**